plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImpl.java

```diff
@@ -908,6 +910,10 @@ private void validateKubernetesClusterUpgradeParameters(UpgradeKubernetesCluster
         if (!KubernetesCluster.State.Running.equals(kubernetesCluster.getState())) {
             throw new InvalidParameterValueException(String.format("Kubernetes cluster ID: %s is not in running state", kubernetesCluster.getUuid()));
         }
+        final DataCenter zone = dataCenterDao.findById(kubernetesCluster.getZoneId());
+        if (zone == null) {
+            logAndThrow(Level.WARN, String.format("Unable to find zone for Kubernetes cluster ID: %s", kubernetesCluster.getUuid()));
```

Comment:
Is `logAndThrow` a new utility?

Reply:
Just some helper methods to log and throw exceptions for similar error https://github.com/apache/cloudstack/blob/8c6861bdc0f6b9fbf58a1c099f36ba08348dd204/plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImpl.java#L216-L249